src/main/java/de/komoot/photon/query/PhotonRequestFactory.java

```diff
@@ -38,7 +40,10 @@ public PhotonRequest create(Request webRequest) throws BadRequestException {
 
         PhotonRequest request = new PhotonRequest(query, languageResolver.resolveRequestedLanguage(webRequest));
 
-        request.setLimit(parseInt(webRequest, "limit"));
+        Integer limit = parseInt(webRequest, "limit");
+        if(limit != null)
+            request.setLimit(Integer.max(Integer.min(limit, maxResults), 1));
```

Comment:
Sorry, style nitpicking: please add a space after 'if' and always curly braces, even for one-liners.